R/apa_print_glht.R

```diff
@@ -58,237 +58,43 @@ apa_print.summary.glht <- function(
 
   tidy_x <- broom::tidy(x)
   test_stat <- ifelse(x$df == 0, "z", paste0("t(", x$df, ")"))
-  conf_level <- paste0(ci * 100, "\\% CI")
+  conf_level <- paste0(ci * 100, "% CI")
+  p_value <- names(tidy_x)[grepl("p.value", names(tidy_x), fixed = TRUE)]
 
   # Assamble table
   ## Add (adjusted) confidence intervall
   multcomp_adjustment <- if(x$test$type == "none") multcomp::univariate_calpha() else multcomp::adjusted_calpha()
   print_ci <- stats::confint(x, level = ci, calpha = multcomp_adjustment)$confint
   dimnames(print_ci) <- NULL
   table_ci <- unlist(print_confint(print_ci[, -1], ...)) # Remove point estimate from matrix
-  contrast_table <- cbind(estimate = tidy_x$estimate, confint = table_ci, tidy_x[, c("statistic", "p.value")])
-  rownames(contrast_table) <- sanitize_terms(tidy_x$lhs)
-
-  ## Format numbers
-  contrast_table$estimate <- printnum(contrast_table$estimate, ...)
-  contrast_table$statistic <- printnum(contrast_table$statistic, digits = 2)
-  contrast_table$p.value <- printp(contrast_table$p.value)
-
-  # Concatenate character strings and return as named list
-  apa_res <- init_apa_results()
-
-  apa_res$estimate <- apply(contrast_table, 1, function(y) {
-    paste0("$\\Delta M = ", y["estimate"], "$, ", conf_level, " ", y["confint"])
-  })
-
-  apa_res$statistic <- apply(contrast_table, 1, function(y) {
-    paste0("$", test_stat, " = ", y["statistic"], "$, $p ", add_equals(y["p.value"]), "$")
-  })
-
-  apa_res$full_result <- paste(apa_res$estimate, apa_res$stat, sep = ", ")
-  names(apa_res$full_result) <- names(apa_res$estimate)
-
-  apa_res <- lapply(apa_res, as.list)
-
-  # Add table
-  rownames(contrast_table) <- tidy_x$lhs
-  colnames(contrast_table) <- c("estimate", "ci", "statistic", "p.value")
-  variable_label(contrast_table) <- c(
-    estimate = "$\\Delta M$"
-    , ci = conf_level
+  tidy_x$std.error <- table_ci
+  colnames(tidy_x)[colnames(tidy_x) == "std.error"] <- "conf.int"
+
+  ## Typeset columns
+  tidy_x$contrast <- sanitize_terms(tidy_x$contrast)
+  tidy_x$estimate <- printnum(tidy_x$estimate, ...)
+  tidy_x$statistic <- printnum(tidy_x$statistic, digits = 2)
+  tidy_x[[p_value]] <- printp(tidy_x[[p_value]])
+
+  ## Add variable labels
+  variable_labels(tidy_x) <- c(
+    contrast = "Contrast"
+    , null.value = "$\\mu_0$"
+    , estimate = "$\\Delta M$"
+    , conf.int = conf_level
     , statistic = paste0("$", test_stat, "$")
-    , p.value = "$p$"
   )
```

Comment:
Would it be desirable to produce a result if any of those standard columns are missing for some reason? That would be an incomplete output, no?

Reply:
I strongly favor to issue a warning, but still return an output in cases when a column is missing (or a column has been renamed by the package authors). This will allow users to continue their work on the manuscript in such cases, but with the warning message issued by `canonize()`, they can still create an informative issue on github. It is my experience that changes to column names were one of the main reasons why `apa_print()` failed in the past (e.g., this bug was introduced because `p.value` was changed to `adj.p.value`. (Another case when this behavior is desirable is when users work with non-standard objects of a supported class, which is something I believe can happen just because the R jungle is large, weakly typed, and dangerous)

Reply:
Agreed. As discussed, we will leave resolving this for a separate PR. Thanks for your feedback!